### DISCUSSION TITLE
TupleQueryResultParserRegistry - Failed to instantiate service

### DISCUSSION THREAD

**Discussion by ToddTam** [eclipse-rdf4j/rdf4j#3346] [Q&A] [answered]
I am very new to RDF4J and GraphDB.  I'm working on one of the simple example programs to connect and run simple queries on a GraphDB database running on my local host.  I'm able to connect and run my simple queries, but I'm getting the following error messages:

[main] ERROR org.eclipse.rdf4j.query.resultio.TupleQueryResultParserRegistry - Failed to instantiate service
java.lang.NoSuchFieldError: JSON_STAR
[main] ERROR org.eclipse.rdf4j.query.resultio.TupleQueryResultParserRegistry - Failed to instantiate service
java.lang.NoSuchFieldError: SPARQL_STAR
[main] ERROR org.eclipse.rdf4j.query.resultio.TupleQueryResultParserRegistry - Failed to instantiate service
java.lang.NoSuchFieldError: TSV_STAR

I'm using Eclipse and followed the steps in the "Starting a New Maven Project in Eclipse" documentation.  From what I can see, I do have the required JAR file and classes in my Maven Dependencies and I can see the definitions of the JSON_STAR, SPARQL_STAR, and TSV_STAR fields.

Does anyone have any suggestions on what the problem might be and how to fix it?

Thank you for your time and consideration.

**Comment by barthanssens** [comment, marked as answer]
Hi,

could you perhaps list the versions of RDF4J you are using locally and the version of the GraphDB server ?
It might be the case that your local version is more recent than the libraries on the GraphDB server

Best regards

Bart

**Reply by ToddTam** [reply]
I'm using RDF4J Version 3.6.0 and GraphDB Free Version 8.8.1 and Java 8.

**Reply by barthanssens** [reply]
Hmz, GraphDB 8.8 uses RDF4J 2.4.2, so that could be the issue (support for RDF-Star was only added in RDF4J 3.2.0).
I assume upgrading to a newer version of GraphDB isn't possible ?

**Reply by ToddTam** [reply]
I'm stuck with GraphDB 8.8.1 for now.  As an experiment, I updated my dependencies from 3.6.0 to 2.4.2, and that does eliminate the error messages.  This is enough to keep me going for now.  Thanks for all of your help.
